### PR TITLE
HIV-618 : Update Patient Registration Details- Level of Education

### PR DESCRIPTION
**Pull request by donaldkibet**

### CODE REVIEW

File: src/app/patient-dashboard/common/patient-info/patient-demographics.component.html [hunk 1]
```diff
@@ -22,7 +22,7 @@
        </tr>
          <tr>
            <td><strong>Age  : </strong> {{patient.person.age}}</td>
-         <td>&nbsp;</td>
+           <td><strong>Highest Educations  : </strong>{{patientLevelOfEducation?.name.display}}</td>
```
Comment (denniskigen) [inline src/app/patient-dashboard/common/patient-info/patient-demographics.component.html]: Maybe ought to be `Highest Education Level`?

Comment (donaldkibet) [reply src/app/patient-dashboard/common/patient-info/patient-demographics.component.html]: :+1:

File: src/app/models/person.model.ts [hunk 2]
```diff
@@ -108,6 +108,20 @@ export class Person extends BaseModel {
       }
     }
   }
+  public get levelOfEducation() {
+    const levelOfEducationPersonAttributeTypeUuid = '2b91b4a5-d421-4a70-bb7c-8adfa083dcef';
+    if (this._attributes.length > 0) {
```
Comment (denniskigen) [inline src/app/models/person.model.ts]: We should check whether `this.attributes` exists first:
`this.attributes && this.attributes.length` or similar.

File: src/app/patient-dashboard/common/patient-info/patient-info.component.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ export class PatientInfoComponent implements OnInit, OnDestroy, AfterViewInit {
   public subscription: Subscription;
   public routeSub: Subscription;
   public scrollSection = '';
+  public levelOfEducation = '';
```
Comment (denniskigen) [inline src/app/patient-dashboard/common/patient-info/patient-info.component.ts]: We don't seem to be using this. Safe to get remove, as well as the `console.log`s.

File: src/app/patient-dashboard/common/patient-info/patient-demograpics.component.ts [hunk 4]
```diff
@@ -40,11 +44,26 @@ export class PatientDemographicsComponent implements OnInit, OnDestroy {
     this.getPatientDemographics();
     this.appFeatureAnalytics
       .trackEvent('Patient Dashboard', 'Lab Orders Loaded', 'ngOnInit');
+    this.patientCreationService
+      .getLevelOfEducation()
+      .subscribe(
+        (res: {
+          answers: Array<{ name: { display: string; uuid: string } }>;
+        }) => {
+          if (!_.isEmpty(res)) {
```
Comment (denniskigen) [inline src/app/patient-dashboard/common/patient-info/patient-demograpics.component.ts]: `res.length` could work here too. Avoids the need for bringing in lodash.

File: src/app/patient-dashboard/common/patient-info/edit-demographics.component.html [hunk 5]
```diff
@@ -94,6 +94,15 @@
       </div>
     </div>
 
+    <div class="col-md-6 col-sm-6 col-lg-6 col-xs-6">
+      <div class="form-group">
+        <label for="levelOfEducation">Highest Education Level</label>
+        <select class="form-control" name="levelOfEducation" [(ngModel)]="patientLevelOfEducation" (ngModelChange)="updateLevelOfEducation($event)" >
+          <option *ngFor="let education of levelOfEducation" [value]="education.name.uuid" >{{education.name.display}}</option>
```
Comment (denniskigen) [inline src/app/patient-dashboard/common/patient-info/edit-demographics.component.html]: Nit: maybe `level` would be the more appropriate variable name.

File: src/app/patient-dashboard/common/patient-banner/patient-banner.component.html [hunk 6]
```diff
@@ -222,4 +222,37 @@ <h4>
 <div class="margin-bottom-5 clear clear-fix">&nbsp;</div>
 
 <add-relationship #addRelationship></add-relationship>
-<edit-identifiers #identifiers></edit-identifiers>
\ No newline at end of file
+<edit-identifiers #identifiers></edit-identifiers>
+
+<ng-template #relationshipTemplate>
```
Comment (denniskigen) [inline src/app/patient-dashboard/common/patient-banner/patient-banner.component.html]: Is this work on relationships meant to be in this PR as well?

**Review by kipchubett — APPROVED** [review]

**Pull request merged**